File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactoryUnitTest.java
package com.mindera.skeletoid.threads.threadpools;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNotNull;
import static org.mockito.Mockito.mock;

public class NamedThreadFactoryUnitTest {

    @Test
    public void testGetThreads() {
        NamedThreadFactory namedThreadFactory = new NamedThreadFactory("name", 10);

        assertNotNull(namedThreadFactory.getThreads());
        assertEquals(0, namedThreadFactory.getThreads().size());
    }

    @Test
    public void testNewThreads() {
        Runnable runnable = mock(Runnable.class);
        NamedThreadFactory namedThreadFactory = new NamedThreadFactory("name", 10);
        Thread thread = namedThreadFactory.newThread(runnable);

        assertNotNull(thread);
        assertEquals(1, namedThreadFactory.getThreads().size());
    }

    @Test
    public void testClearThreads() {
        NamedThreadFactory namedThreadFactory = new NamedThreadFactory("name", 10);

        namedThreadFactory.clearThreads();
        assertNotNull(namedThreadFactory.getThreads());
        assertEquals(0, namedThreadFactory.getThreads().size());
    }
}

File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactoryUnitTest.kt
package com.mindera.skeletoid.threads.threadpools

import junit.framework.Assert
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito

class NamedThreadFactoryUnitTest {

    lateinit var namedThreadFactory : NamedThreadFactory

    @Before
    fun setUp() {
        namedThreadFactory = NamedThreadFactory("name", 10)
    }

    @Test
    fun testGetThreads() {
        Assert.assertNotNull(namedThreadFactory.threads)
        Assert.assertEquals(0, namedThreadFactory.threads.size)
    }

    @Test
    fun testNewThreads() {
        val runnable =
            Mockito.mock(Runnable::class.java)
        val thread = namedThreadFactory.newThread(runnable)
        Assert.assertNotNull(thread)
        Assert.assertEquals(1, namedThreadFactory.threads.size)
    }

    @Test
    fun testClearThreads() {
        namedThreadFactory.clearThreads()
        Assert.assertNotNull(namedThreadFactory.threads)
        Assert.assertEquals(0, namedThreadFactory.threads.size)
    }
}